dist/dfu.js

```diff
@@ -253,103 +309,110 @@
-                        .then(() => {
-                            return packetChar.writeValue(new Uint8Array([0]));
-                        })
-                        .then(() => {
-                            return controlChar.writeValue(new Uint8Array([2,1]));
-                        })
-                        .catch(error => {
-                            error = "error writing init: " + error;
-                            log(error);
-                            reject(error);
-                        });
-
-                    } else if (req_opcode === 1 || req_opcode === 2) {
-                        log('complete, send packet count');
-
-                        var buffer = new ArrayBuffer(3);
-                        view = new DataView(buffer);
-                        view.setUint8(0, 8);
-                        view.setUint16(1, interval, littleEndian);
-
-                        controlChar.writeValue(view)
-                        .then(() => {
-                            log("sent packet count: " + interval);
-                            return controlChar.writeValue(new Uint8Array([3]));
-                        })
-                        .then(() => {
-                            log("sent receive");
-                            return writePacket(packetChar, arrayBuffer, 0);
-                        })
-                        .catch(error => {
-                            error = "error sending packet count: " + error;
-                            log(error);
-                            reject(error);
-                        });
-
-                    } else if (req_opcode === 3) {
-                        log('complete, check length');
-
-                        controlChar.writeValue(new Uint8Array([7]))
-                        .catch(error => {
-                            error = "error checking length: " + error;
-                            log(error);
-                            reject(error);
-                        });
-
-                    } else if (req_opcode === 7) {
-                        var byteCount = view.getUint32(3, littleEndian);
-                        log('length: ' + byteCount);
-                        log('complete, validate...');
-
-                        controlChar.writeValue(new Uint8Array([4]))
-                        .catch(error => {
-                            error = "error validating: " + error;
-                            log(error);
-                            reject(error);
-                        });
-
-                    } else if (req_opcode === 4) {
-                        log('complete, reset...');
+                    
+                    switch(req_opcode) {
+                        case OPCODE.START_DFU:
+                            log('write init packet');
+    
+                            if(majorVersion <= 6) { // init packet is not used in SDK v6 (so not used in mbed).
+                              break;
+                            }
```

Comment:
I can't see how this works with version <=6, it will just drop out and not initiate the transfer. I may be wrong, but that's the reason I didn't use a switch.